src/main/java/seedu/address/model/scheduler/time/TimePeriod.java

```diff
@@ -94,6 +97,14 @@ public Day getSchoolDay() {
         return schoolDay;
     }
 
+    public List<HourBlock> fragmentIntoHourBlocks() {
+        List<HourBlock> hourBlocks = new ArrayList<>();
+        for (int hour = getStartTime().getHourOfDay(); hour < getEndTime().getHourOfDay(); hour++) {
+            hourBlocks.add(new HourBlock(new LocalTime(hour, 0), schoolDay));
+        }
+        return hourBlocks;
+    }
```

Comment:
Can there be a toString? Something like "[day] start time - end time"

Reply:
Hi. Thats my department. `TimePeriod` is abstract so it cannot be instantiated. So object passed will either be `HourBlock` or `TimeBlock` in compile time. They have their own toString methods already.

Reply:
Alright thanks!